packages/components/pie-tabs/src/index.ts

```diff
@@ -0,0 +1,132 @@
+import { html, unsafeCSS } from 'lit';
+import { PieElement } from '@justeattakeaway/pie-webc-core/src/internals/PieElement';
+import { safeCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
+import { classMap } from 'lit/directives/class-map.js';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+
+import styles from './tabs.scss?inline';
+import {
+    type TabsProps,
+    variants,
+    defaultProps,
+    orientations,
+} from './defs';
+import { type TabPanelProps } from './pie-tab-panel/defs';
+
+const componentSelector = 'pie-tabs';
+
+// Valid values available to consumers
+export * from './defs';
+
+/**
+ * @tagname pie-tabs
+ */
+@safeCustomElement(componentSelector)
+export class PieTabs extends PieElement implements TabsProps {
+    @property({ type: String })
+    @validPropertyValues(componentSelector, variants, defaultProps.variant)
+    public variant = defaultProps.variant;
+
+    @property({ type: String })
+    @validPropertyValues(componentSelector, orientations, defaultProps.orientation)
+    public orientation = defaultProps.orientation;
+
+    @queryAssignedElements() _pieTabPanelSlots!: Array<HTMLElement & TabPanelProps>;
+
+    private _selectedTab = 0;
+
+    firstUpdated (): void {
+        this.requestUpdate();
+    }
+
+    updated (): void {
+        this.updateSelectedPanel();
+    }
+
+    /**
+     * Handles the click event on a tab.
+     * This method updates the selected tab index and updates the displayed panel accordingly.
+     *
+     * @param index The index of the tab that was clicked.
+     *
+     * @private
+     */
+    private handleTabClick (index: number) {
+        this._selectedTab = index;
+        this.updateSelectedPanel();
+        this.requestUpdate();
+    }
+
+    /**
+     * Updates the selected state of each tab panel based on the currently selected tab index.
+     * This method iterates through all tab panels and sets the `selected` property accordingly.
+     *
+     * @private
+     */
+    private updateSelectedPanel () {
+        this._pieTabPanelSlots.forEach((panel, index) => {
+            panel.selected = index === this._selectedTab;
+        });
+    }
+
+    render () {
+        const classes = {
+            'c-tabs': true,
+            'c-tabs-variant--contained': this.variant === 'contained',
+            [`c-tabs-orientation--${this.orientation}`]: true,
+        };
+
+        return html`
+            <div
+                data-test-id="pie-tabs"
```

Comment:
Not fully sure how is the approach on other components, but we should be able to pass different data-testid's to the components. 

Could we add this as a prop, with a default value of `pie-tabs`?